### DISCUSSION TITLE
Use a slug field for ModelChoiceFilter

### DISCUSSION THREAD

**Discussion by CelestialGuru** [carltongibson/django-filter#1344] [Q&A] [answered]
I have a model which has a foreign key to an enum table. I'd rather filter by the enum name rather than the id.

Product:
id | name | quality
-|-|-
50 | Chair | 3
51 | Pen | 1

Quality:
id | name
-|-
1 | poor
2 | okay
3| good

I'd like to filter with something like `products/?quality=poor` instead of `products/?quality=1`

## My current solution is this:
```python
class ProductFilterSet(FilterSet):
    class Meta:
        model = Product
        fields = ("quality",)

    quality= CharFilter(field_name="quality", method="filter_quality")

    def filter_quality(self, queryset, name, value):
        return queryset.filter(**{f"{name}__name": value})
```
## I want something like DRF's `SlugRelatedField`
You'd have something like
```python
class ProductFilterSet(FilterSet):
    class Meta:
        model = Product
        fields = ("quality",)

    quality= SlugRelatedFilter(slug_field="name", queryset=Quality.objects.all())
```
## Bonus round
Oh, and for bonus points, could we have it where you can alias the field too? Handy when you rename fields and your api needs to stay the same. The code might look something like
```python
class ProductFilterSet(FilterSet):
    class Meta:
        model = Product
        fields = ("product_quality",)

    product_quality = SlugRelatedFilter(slug_field="name", source="quality", queryset=Quality.objects.all())
```

---

Maybe it's possible for me to write my own, but I'm just not that familiar with how this library works. Any suggestions?

**Comment by carltongibson** [comment, marked as answer]
You should be able to use `to_field_name`: https://django-filter.readthedocs.io/en/stable/ref/filters.html#to-field-name

**Reply by CelestialGuru** [reply]
Okay, for posterity's sake, this is the solution:
```python
class YourFilterSet(FilterSet):
   class Meta:
      model = YourModel
      fields = (
         "url_string", # this is what you want in the url like model/?url_string=whatever
      )

   url_string = ModelChoiceFilter(
      field_name="name_of_fk_field_on_first_model", # doesn't have to match what you want in the url querystring key
      to_field_name="field_on_other_table_to_use_instead_of_primary_key", # slug field
      queryset=OtherModel.objects.all(),
   )
```
